src/components/Utils/hooks.tsx

```diff
@@ -1,15 +1,18 @@
 import React from 'react'
 
-function useSafeDispatch(dispatch) {
+function useSafeDispatch(dispatch: React.Dispatch<IAction>) {
   const mounted = React.useRef(false)
 
   React.useLayoutEffect(() => {
     mounted.current = true
-    return () => (mounted.current = false)
+    return () => {
+      mounted.current = false
+    }
   }, [])
 
   return React.useCallback(
-    (...args) => (mounted.current ? dispatch(...args) : void 0),
+    ({...args}): React.Dispatch<IAction> | void =>
+      mounted.current ? dispatch({...args}) : void 0,
```

Comment:
`...args` for https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters is a different feature than the object spead `{...args}` - please don't convert